apps/backend/src/modules/contributors/contributor.repository.ts

```diff
@@ -15,35 +17,68 @@ class ContributorRepository implements Repository {
 
 		const contributor = await this.contributorModel
 			.query()
-			.insert({
-				name,
-			})
+			.insert({ name })
 			.execute();
 
-		return ContributorEntity.initialize(contributor);
+		return ContributorEntity.initialize({ ...contributor, projects: [] });
 	}
 
 	public delete(): ReturnType<Repository["delete"]> {
 		return Promise.resolve(true);
 	}
 
 	public async find(id: number): Promise<ContributorEntity | null> {
-		const item = await this.contributorModel.query().findById(id).execute();
+		const contributor = await this.contributorModel
+			.query()
+			.findById(id)
+			.withGraphFetched("gitEmails");
 
-		return item ? ContributorEntity.initialize(item) : null;
+		if (!contributor) {
+			return null;
+		}
+
+		return ContributorEntity.initialize({ ...contributor, projects: [] });
 	}
 
-	public findAll(): ReturnType<Repository["findAll"]> {
-		return Promise.resolve({ items: [] });
+	public async findAll(): Promise<{ items: ContributorEntity[] }> {
+		const contributorsWithProjectsAndEmails = await this.contributorModel
+			.query()
+			.select("contributors.*")
+			.select(
+				raw(
+					"COALESCE(ARRAY_AGG(DISTINCT jsonb_build_object('id', projects.id, 'name', projects.name)) FILTER (WHERE projects.id IS NOT NULL), '{}') AS projects",
+				),
+			)
+			.select(
+				raw(
+					"COALESCE(ARRAY_AGG(DISTINCT jsonb_build_object('id', git_emails.id, 'email', git_emails.email)) FILTER (WHERE git_emails.id IS NOT NULL), '{}') AS git_emails",
+				),
+			)
```

Comment:
I think we can join git emails the usual way, with relation mapping, because we will probably need this in another place